File: ProtocolSharp/Types/Identifier.cs
using System;

namespace ProtocolSharp.Types
{
	public class Identifier
	{
		public string Namespace { get; set; } = "minecraft";

		public string Name { get; set; }

		public override string ToString()
		{
			return $"{Namespace}:{Name}";
		}

		public Identifier(string ns, string nm)
		{
			Namespace = ns ?? "minecraft";
			Name = nm ?? throw new ArgumentNullException(nameof(nm));
		}

		public Identifier(string name)
		{
			Name = name ?? throw new ArgumentNullException(nameof(name));
		}
	}
}

File: Minecraft Types/src/com/CopperPenguin96/MinecraftTypes/Identifier.java
package com.CopperPenguin96.MinecraftTypes;

public class Identifier {
	public String Namespace = "minecraft";
	public String Name;
	
	@Override
	public String toString() {
		if (Namespace.isEmpty()) {
			Namespace = "minecraft";
		}
		
		return Namespace + ":" + Name;
	}
	
	public Identifier(String namespace, String name) {
		Namespace = namespace;
		Name = name;
	}
	
	public Identifier(String name) {
		this("minecraft", name);
	}
}
